service/application/src/main/java/com/solace/maas/ep/event/management/agent/service/ScanService.java

```diff
@@ -449,4 +451,22 @@ protected RouteBundleHierarchyStore registerRouteRecipients(RouteBundle routeBun
         }
         return pathStore;
     }
+
+    public boolean isScanComplete(String scanId) {
+        Set<String> completeScanStatuses = Set.of(
```

Comment:
Public method, so we SHOULD validate the input parameter scanId, check ifnull or empty

Reply:
same pattern applied as in e.g. findByMessagingServiceId(...) - if there is no scanId the scanTypeRepository will throw an exception